core/src/main/resources/org/apache/spark/ui/static/table.js

```diff
@@ -30,3 +30,75 @@ function stripeSummaryTable() {
        }
     });
 }
+
+function toggleThreadStackTrace(threadId, forceAdd) {
+    var stackTrace = $("#" + threadId + "_stacktrace")
+    if (stackTrace.length == 0) {
+        var stackTraceText = $('#' + threadId + "_td_stacktrace").html()
+        var threadCell = $("#thread_" + threadId + "_tr")
+        threadCell.after("<tr id=\"" + threadId +"_stacktrace\" class=\"accordion-body\"><td colspan=\"3\"><pre>" +
+            stackTraceText +  "</pre></td></tr>")
+    } else {
+        if (!forceAdd) {
+            stackTrace.remove()
+        }
+    }
+}
+
+// expandOrCollapse - true: expand, false: collapse
+function expandOrCollapseAllThreadStackTrace(expandOrCollapse, toggleButton) {
+    if (expandOrCollapse) {
+        $('.accordion-heading').each(function() {
+            //get thread ID
+            if (!$(this).hasClass("hidden")) {
+                var trId = $(this).attr('id').match(/thread_([0-9]+)_tr/m)[1]
+                toggleThreadStackTrace(trId, true)
+            }
+        })
+        if (toggleButton) {
+            $('.expandbutton').toggleClass('hidden')
+        }
+    } else {
+        $('.accordion-body').each(function() {
+            $(this).remove()
+        })
+        if (toggleButton) {
+            $('.expandbutton').toggleClass('hidden');
+        }
+    }
+}
+
+// inOrOut - true: over, false: out
+function onMouseOverAndOut(threadId) {
+    $("#" + threadId + "_td_id").toggleClass("threaddump-td-mouseover");
+    $("#" + threadId + "_td_name").toggleClass("threaddump-td-mouseover");
+    $("#" + threadId + "_td_state").toggleClass("threaddump-td-mouseover");
+}
+
+function onSearchStringChange() {
+    var searchString = $('#search').val()
```

Comment:
add a `toLowerCase` here -- I tried searching for "RUNNABLE" first and it didn't find anything.